static/app/components/nav/overlay.tsx

```diff
@@ -0,0 +1,60 @@
+import {type ComponentProps, forwardRef} from 'react';
```

Comment:
Just as a word of caution, but whenever we use ComponentProps<typeof Component>, we end up creating a unnecessary type alias with typeof Component, which over time will slow down the type checking and increase memory usage of the language server. Since we dont really have a convention, this is obv fine, but we should really just be importing the component props directly

Reply:
Good call! I will definitely keep that in mind. I refactored another usage in this area in 7c73aeaf47d